test/cmake/platformenum.cpp

```diff
@@ -5,7 +5,7 @@
 #include <iostream> // std::cout
 #include <cstdlib> // EXIT_FAILURE
 
-int main()
+int main(void)
```

Comment:
I understand this change cause semantic differences in C file, it's not to clear what difference this change makes in C++ however. Did something fail without it?

Reply:
This is an old change so I'm not fully sure, but I _think_ it triggered some CI build failures when testing the Windows build with clangcl